text/dependencies/rfcs/0000-dependency-management-overview.md

```diff
@@ -0,0 +1,197 @@
+   The workflow is triggered by the Get New Versions workflow when a new
+   version is discovered. The build process in step 3 is delegated to
+   [cloudfoundry/buildpacks-ci](https://github.com/cloudfoundry/buildpacks-ci/tree/384c051f48fdb4b40521daaacc6afaab87da3796/tasks/build-binary-new)
+   and
+   [cloudfoundry/binary-builder](https://github.com/cloudfoundry/binary-builder/tree/main/recipe)
+   and may involve pulling from source, compiling, or processing the dependency
+   in some way. The code is written in Ruby.  The compiled dependency is
+   uploaded to an S3 bucket. The workflow also gathers metadata (CPEs,
+   licenses, SHA256, etc.) about the dependency using dep-server
+   dependency-specific code.
+4. [Test and Upload
+   Workflow](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/test-upload-metadata.yml).
+   The workflow is triggered by the Build and Upload workflow and receives the
+   metadata from that step. A smoke test is run against the compiled
+   dependency, and if successful, the dependency metadata is uploaded to an AWS
+   S3 bucket.
+5. Endpoint routing for the dep-server is set up through AWS Route 53 and
+   Cloudfront to access dependencies and metadata, but the actual dep-server
+   runs in Google App Engine.
+6. [Update dependencies
+   Workflow](https://github.com/paketo-buildpacks/github-config/blob/main/implementation/.github/workflows/update-dependencies.yml).
+   Each dependency-providing buildpack has a workflow that runs the `jam
+   update-dependencies` command on a timer (or when manually triggered), which
+   will update dependency versions from the dep-server depending on
+   `[[metadata.dependency-constraints]]` listed in the `buildpack.toml`.
+</details>
+
+## Plan Overview
+
+The plan overview provides high level direction for a new dependency management
+approach, split into phases that will help to address the concerns cited in
+the Motivation section of the RFC. Due to the complex nature of the changes,
+the proposals details will be split into separate RFCs.
+
+The new system will focus on shifting all dependency-specific logic into the
+buildpacks themselves. Related automation to actually update and build the
+dependencies will be generic, so that our Github Actions can simply leverage the
+dependency-specific logic in the buildpacks. This allows for buildpack
+maintainers to take ownership on where dependencies come from, if and how
+they're compiled, and the ability to manage all related automation themselves.
+Relatedly, dependencies maintainers will be responsible for setting up general
+project-level dependency concerns like automation, best
+practices, and maintaining legacy dependencies in the dep-server.
+
+The other focus is to enable Stack flexibility by moving away from supporting
+only Ubuntu 18.04 dependencies, which is what we have now due to the way
+dependencies are compiled. While some dependencies may still need to be
```

Comment:
Is this really a compilation problem or is it a metadata problem?

For example our compilation of `node` runs fine on jammy apparently
```
docker run -it -v /tmp/paketo-node:/node ubuntu:jammy bash
root@eaa064b73682:/# cd node
root@eaa064b73682:/node# tar xzf node_v14.19.2_linux_x64_bionic_3f263e3e.tgz
root@eaa064b73682:/node# bin/node --help
Usage: node [options] [ script.js ] [arguments]
       node inspect [options] [ script.js | host:port ] [arguments]
```

And indeed dynamically links against the same (presumably stable) set of glilbc APIs as the upstream distribution
```
cnb@0e9092f382aa:/workspace$ ldd /layers/paketo-buildpacks_node-engine/node/bin/node
	linux-vdso.so.1 (0x00007ffd61cd1000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f4729087000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f4728cfe000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f4728960000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f4728748000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f4728529000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f4728138000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f472dc00000)
```

In cases where the results of compilation differ between stacks (e.g. nodejs for alpine or `nginx` probably) we will still need multiple version per stack even if we fetch binaries from from upstream.

Not suggesting that we shouldn't move away from recompiling, but that in some cases we could add support for more stacks with our current set of deps.